tidy3d/components/geometry.py

```diff
@@ -511,6 +511,28 @@ def reflect_points(
 
         return points_new
 
+    @abstractmethod
+    def volume(self, bounds: Bound) -> float:
+        """Returns object's volume. If the object extends beyound bounds then
+        the computed volume may be only a rough approximation
```

Comment:
Period after the main text ("approximation.") and yea add `Parameters` section too please.